app/forms/recovery_code_form.rb

```diff
@@ -21,6 +21,8 @@ def submit
   attr_reader :user, :success
 
   def valid_recovery_code?
+    word_regexp = /\w{#{RandomPhrase::WORD_LENGTH}}/
+    return false unless code =~ /\A#{word_regexp} #{word_regexp} #{word_regexp} #{word_regexp}\Z/
```

Comment:
The way to test this is to verify that `RecoveryCodeGenerator` does not receive `:new` when the code format is invalid. See https://github.com/18F/identity-idp/pull/1251 as an example.